### PR TITLE
main components added

### PR DESCRIPTION
**Pull request by ufoBeliever**
main components added

### CODE REVIEW

File: src/components/DropDown/index.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import styles from "./DropDown.module.scss";
+import { IDropDownProps } from "./types";
+const DropDown: React.FC<IDropDownProps> = ({ items }) => {
+  return (
+    <select className={styles["drop-down"]}>
+      {items.map((e, i) => {
+        return <option key={i}>{e}</option>;
+      })}
```
Comment (igorizviekov) [inline src/components/DropDown/index.tsx:8]: refactor it to remove `return`

File: src/components/DropDown/types.d.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export interface IDropDownProps {
+  items: Array<any>;
```
Comment (igorizviekov) [inline src/components/DropDown/types.d.ts:2]: can we type it more strict

File: src/components/Footer/Footer.module.scss [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  }
+
+  &__column {
+    display: flex;
+    flex-direction: column;
+    width: 350px;
+    gap: 14px;
+  }
+
+  &__column-title {
```
Comment (igorizviekov) [inline src/components/Footer/Footer.module.scss:39]: it should be nested

File: src/components/Header/Header.module.scss [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+  &__add {
+    background: #52106a;
+    height: 42px;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    text-align: center;
+    padding: 0 15px;
+
+    h4 {
```
Comment (igorizviekov) [inline src/components/Header/Header.module.scss:18]: add class to not add styles to an empty tag

File: src/components/Header/Header.module.scss [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+  .header {
+    &__buttons {
+      button:nth-child(2) {
+        display: none;
+      }
+    }
+  }
+}
+
+@media screen and (max-width: 520px) {
```
Comment (igorizviekov) [inline src/components/Header/Header.module.scss:122]: use mixins in css query

Comment (igorizviekov) [reply src/components/Header/Header.module.scss:122]: stick css query to the design mockups f.e. 360/768

File: src/components/Header/Header.module.scss [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+  .header {
+    &__content {
+      margin: 0 15px;
+      height: 104px;
+    }
+    &__add {
+      height: 44px;
+
+      h4 {
```
Comment (igorizviekov) [inline src/components/Header/Header.module.scss:131]: do not add style to an empty tag

File: src/components/Header/Header.module.scss [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+        font-weight: 500;
```
Comment (igorizviekov) [inline src/components/Header/Header.module.scss:132]: create a var

File: src/components/Input/index.tsx [hunk 8]
```diff
@@ -0,0 +1,7 @@
+import styles from "./Input.module.scss";
+
+const Input = () => {
+  return <input className={styles["input"]} />;
```
Comment (igorizviekov) [inline src/components/Input/index.tsx:4]: refactor it to remove `return`

File: src/components/PaginationButton/index.tsx [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+  label,
+  isChecked,
+}) => {
+  return (
+    <button
+      className={
+        styles["pagination-button"] +
+        " " +
+        (isChecked ? styles["pagination-button_checked"] : "")
+      }
```
Comment (igorizviekov) [inline src/components/PaginationButton/index.tsx:14]: refactor classnames

**Review by igorizviekov — CHANGES_REQUESTED** [review]

File: src/components/Button/Button.module.scss [hunk 10]
```diff
@@ -0,0 +1,29 @@
+.button {
+  font-family: "Avenir Next";
+  font-style: normal;
```
[the remark this answers is not in the document]
Comment (ufoBeliever) [reply src/components/Button/Button.module.scss:2]: https://github.com/igorizviekov/light-up/pull/7/commits/5e8166ed9f7ab2511ec9878c294120f1496ef27c

File: src/components/Button/Button.module.scss [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+  align-items: center;
+  justify-content: center;
+
+  width: min-content;
+
+  &_secondary {
+    background-color: white;
+    box-sizing: border-box;
+    border: 1px #52106a solid;
+    color: #52106a;
```
[the remark this answers is not in the document]
Comment (ufoBeliever) [reply src/components/Button/Button.module.scss]: https://github.com/igorizviekov/light-up/pull/7/commits/293d329506310e66cb9993450c77f30225e590e0

File: src/components/Button/Button.module.scss [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+  padding: 14px 30px;
+
+  display: flex;
+  align-items: center;
+  justify-content: center;
+
+  width: min-content;
+
+  &_secondary {
+    background-color: white;
```
[the remark this answers is not in the document]
Comment (ufoBeliever) [reply src/components/Button/Button.module.scss]: https://github.com/igorizviekov/light-up/pull/7/commits/293d329506310e66cb9993450c77f30225e590e0

File: src/components/Button/Button.module.scss [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+  font-style: normal;
+  font-weight: 500;
+  font-size: 12px;
+  line-height: 16px;
+
+  text-align: center;
+  letter-spacing: 2.4px;
+  text-transform: uppercase;
+
+  color: #ffffff;
```
[the remark this answers is not in the document]
Comment (ufoBeliever) [reply src/components/Button/Button.module.scss]: https://github.com/igorizviekov/light-up/pull/7/commits/293d329506310e66cb9993450c77f30225e590e0

File: src/components/DropDown/index.tsx (hunk 1 above)
Comment (ufoBeliever) [reply src/components/DropDown/index.tsx]: https://github.com/igorizviekov/light-up/pull/7/commits/794e3f1b32505b24dd56d4254d2c216adbb5ee4d

File: src/components/Input/index.tsx (hunk 8 above)
Comment (ufoBeliever) [reply src/components/Input/index.tsx]: https://github.com/igorizviekov/light-up/pull/7/commits/794e3f1b32505b24dd56d4254d2c216adbb5ee4d

File: src/components/DropDown/DropDown.module.scss [hunk 14]
```diff
@@ -0,0 +1,18 @@
+.drop-down {
+  box-sizing: border-box;
+  width: 60px;
+  height: 35px;
+  background: #ffffff;
+  border: 0.754526px solid #c4c4c4;
+  border-radius: 4.5px;
+
+  font-family: "Avenir Next";
```
[the remark this answers is not in the document]
Comment (ufoBeliever) [reply src/components/DropDown/DropDown.module.scss]: https://github.com/igorizviekov/light-up/commit/5e8166ed9f7ab2511ec9878c294120f1496ef27c

File: src/components/DropDown/types.d.ts (hunk 2 above)
Comment (ufoBeliever) [reply src/components/DropDown/types.d.ts]: https://github.com/igorizviekov/light-up/pull/7/commits/1a5d2c1645246fff4574c2c630689f33b3316295

File: src/components/PaginationButton/index.tsx (hunk 9 above)
Comment (ufoBeliever) [reply src/components/PaginationButton/index.tsx]: https://github.com/igorizviekov/light-up/pull/7/commits/6b69e99e000e913866c2c5cbd9c26a5fc0376aec

File: src/components/Button/index.tsx [hunk 15]
```diff
@@ -0,0 +1,17 @@
+import styles from "./Button.module.scss";
+import { IButtonProps } from "./type";
+const Button: React.FC<IButtonProps> = ({ type, children }) => {
+  return (
+    <button
+      className={
+        styles["button"] +
+        " " +
+        (type === "secondary" ? styles["button_secondary"] : "")
+      }
```
[the remark this answers is not in the document]
Comment (ufoBeliever) [reply src/components/Button/index.tsx]: https://github.com/igorizviekov/light-up/pull/7/commits/6b69e99e000e913866c2c5cbd9c26a5fc0376aec

File: src/components/Header/Header.module.scss (hunk 7 above)
Comment (ufoBeliever) [reply src/components/Header/Header.module.scss]: https://github.com/igorizviekov/light-up/pull/7/commits/e1b8f39f5217f487124296a507c3e31c860d8306

File: src/components/Header/Header.module.scss (hunk 6 above)
Comment (ufoBeliever) [reply src/components/Header/Header.module.scss]: https://github.com/igorizviekov/light-up/pull/7/commits/e1b8f39f5217f487124296a507c3e31c860d8306

File: src/components/Header/Header.module.scss (hunk 4 above)
Comment (ufoBeliever) [reply src/components/Header/Header.module.scss]: https://github.com/igorizviekov/light-up/pull/7/commits/e1b8f39f5217f487124296a507c3e31c860d8306

File: src/components/Footer/Footer.module.scss (hunk 3 above)
Comment (ufoBeliever) [reply src/components/Footer/Footer.module.scss]: https://github.com/igorizviekov/light-up/pull/7/commits/e1b8f39f5217f487124296a507c3e31c860d8306

File: src/components/Button/index.tsx (hunk 15 above)
[the remark this answers is not in the document]
Comment (igorizviekov) [reply src/components/Button/index.tsx]: classNames issue

**Review by igorizviekov — COMMENTED** [review]
Try to be more descriptive in commit messages
image

File: src/components/Header/Header.module.scss [hunk 16]
```diff
@@ -0,0 +1,160 @@
...
+  &__add {
+    background: var(--primary);
+    height: 42px;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    text-align: center;
+    padding: 0 15px;
+
+    &-text {
```
Comment (igorizviekov) [inline src/components/Header/Header.module.scss:18]: Please refactor it to be `&__text` or even something more descriptive.

Use `--` when you style div states f.e. `&--active` `&--hover`

File: src/components/Header/Header.module.scss [hunk 17]
```diff
@@ -0,0 +1,160 @@
...
+  }
+
+  &__line {
+    width: 100%;
+    height: 2px;
+    background: var(--gray-9);
+  }
+}
+
+@media screen and (max-width: 1440px) {
```
Comment (igorizviekov) [inline src/components/Header/Header.module.scss:78]: Please do it as scss mixin. Refactor to match media queries of the Figma mockup

File: src/components/PaginationButton/index.tsx [hunk 18]
```diff
@@ -0,0 +1,20 @@
...
+const PaginationButton: React.FC<IPaginationButtonProps> = ({
+  label,
+  isChecked,
+}) => {
+  return (
+    <button
+      className={[
+        styles["pagination-button"],
+        isChecked && styles["pagination-button_checked"],
+      ].join(" ")}
```
Comment (igorizviekov) [inline src/components/PaginationButton/index.tsx:13]: Please make a utility function for it. Follow this approach when copying/pasting the code a lot. Or at least remove it from JSX by creating a const in your component

**Review by igorizviekov — CHANGES_REQUESTED** [review]

File: src/components/Header/Header.module.scss (hunk 16 above)
Comment (ufoBeliever) [reply src/components/Header/Header.module.scss:18]: but it is not suitable for BEM rules

File: src/components/Header/Header.module.scss (hunk 17 above)
Comment (ufoBeliever) [reply src/components/Header/Header.module.scss:78]: in some cases horizontal scroll appears, so I should set another media queries' width

**Review by igorizviekov — APPROVED** [review]

**Pull request merged**